lms/djangoapps/discussion/rest_api/permissions.py

```diff
@@ -89,8 +89,14 @@ def get_editable_fields(cc_content: Union[Thread, Comment], context: Dict) -> Se
     is_thread = cc_content["type"] == "thread"
     is_comment = cc_content["type"] == "comment"
     is_privileged = context["is_requester_privileged"]
-    # True if we're dealing with a closed thread or a comment in a closed thread
-    is_thread_closed = cc_content["closed"] if is_thread else context["thread"]["closed"]
+
+    if is_thread:
+        is_thread_closed = cc_content["closed"]
+    elif context.get("thread"):
+        is_thread_closed = context["thread"]["closed"]
+    else:
+        # No editable fields when outside thread context
+        return set()
```

Comment:
What's the use case here? Is it because of "comment" but I suspect comment is also a type of a "thread" but I may be wrong?

Reply:
The issue here is that the existing code is designed to deal with comments in the context of a thread. So when you fetch a set of comments for a thread, you can load the thread, and use this information to determine which fields are editable. For instance, a comment on a closed thread, it cannot be edited or endorsed etc. 

However, if you're fetching comments that could be from a lot of different threads, you will either needs to load dozens of threads and recompute these for each one of them, or you can just skip this and disable this information when you're fetching thread based on a user instead of a thread.